staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/compilation_test.go

```diff
@@ -0,0 +1,511 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cel
+
+import (
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions"
+	"k8s.io/apiextensions-apiserver/pkg/apiserver/schema"
+	"strings"
+	"testing"
+)
+
+type validationMatch struct {
```

Comment:
Nit: use `cmp` and `cmpopts` to do the comparisons.

Reply:
cmp compare for string is also using strings.contains. And we want to check the error contains specific message instead of doing diff.

Reply:
All of that is achievable with `cmpopts`. The suggestion is to do something like:

```golang

cases := []struct {
  wantErrors []*Error
}


diff := cmp.Diff(tc.wantErrors, gotErrors)
```

gotErrors would include all the values coming from `[]CompilationResult` (including `nil`)